### PR TITLE
fixed the poller count metric

### PR DESCRIPTION
**Pull request by mkolodezny**
**What changed?**
clean the expired pollers before reporting the metric

**Why?**
follow up on https://github.com/uber/cadence/pull/3918

**How did you test it?**
tested locally

### CODE REVIEW

File: service/matching/pollerHistory.go [hunk 1]
```diff
@@ -68,6 +74,7 @@ func (pollers *pollerHistory) updatePollerInfo(id pollerIdentity, ratePerSecond
 		rps = *ratePerSecond
 	}
 	pollers.history.Put(id, &pollerInfo{ratePerSecond: rps})
+	pollers.onHistoryUpdatedFunc()
```
Comment (mantas-sidlauskas) [inline service/matching/pollerHistory.go:77]: nit: do we need to check for a nil here?

File: service/matching/taskListManager.go [hunk 2]
```diff
@@ -174,8 +174,9 @@ func newTaskListManager(
 	} else {
 		taskListTypeMetricScope = tlMgr.metricScope().Tagged(taskListDecisionTypeTag)
 	}
-	tlMgr.pollerHistory = newPollerHistory(func(state *cache.UpdatedState) {
-		taskListTypeMetricScope.UpdateGauge(metrics.PollerPerTaskListCounter, float64(state.NewSize))
+	tlMgr.pollerHistory = newPollerHistory(func() {
+		taskListTypeMetricScope.UpdateGauge(metrics.PollerPerTaskListCounter,
+			float64(len(tlMgr.pollerHistory.getAllPollerInfo())))
```
Comment (mantas-sidlauskas) [inline service/matching/taskListManager.go:179]: sadly we can't (?) use history.Size() here

**Review by mantas-sidlauskas — APPROVED** [review]
smgt

File: service/matching/taskListManager.go (hunk 2 above)
Comment (mkolodezny) [reply service/matching/taskListManager.go:179]: yes, as discussed offline, for current cache implementation we need to evict the expired pollers everytime now.

**Pull request merged**